api/v1alpha1/envoyproxy_types.go

```diff
@@ -93,6 +93,13 @@ type EnvoyProxySpec struct {
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
 
+	// EnableZoneDiscovery defines whether to enable service zone discovery via
+	// topology.kubernetes.io/zone label on the underlying node
+	// Default: false
+	//
+	// +optional
+	EnableZoneDiscovery *bool `json:"enableZoneDiscovery,omitempty"`
```

Comment:
cc @envoyproxy/gateway-maintainers / @envoyproxy/gateway-reviewers to help with naming here

also this should probably be a struct and an empty struct to being with, so [minEndpointsPerZone](https://kubernetes.io/docs/concepts/services-networking/topology-aware-routing/#three-or-more-endpoints-per-zone) like settings can be added in the future

Reply:
I think something like minEndpointsPerZone should live under load balancer settings personally since it would be configured per-backend rather than globally.

This field would solely be for enabling the feature in EnvoyProxy.